tests/foreman/api/test_activationkey.py

```diff
@@ -50,470 +50,482 @@ def _bad_max_hosts():
+
+    :expectedresults:
+
+        1. By default, an activation key is associated with no host
+           collections.
+        2. After associating an activation key with some set of host
+           collections and reading that activation key, the correct host
+           collections are listed.
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    # An activation key has no host collections by default.
+    act_key = entities.ActivationKey(organization=module_org).create()
+    assert len(act_key.host_collection) == 0
+
+    # Give activation key one host collection.
+    act_key.host_collection.append(entities.HostCollection(organization=module_org).create())
+    act_key = act_key.update(['host_collection'])
+    assert len(act_key.host_collection) == 1
+
+    # Give activation key second host collection.
+    act_key.host_collection.append(entities.HostCollection(organization=module_org).create())
+    act_key = act_key.update(['host_collection'])
+    assert len(act_key.host_collection) == 2
+
+
+@pytest.mark.tier2
+@pytest.mark.upgrade
+def test_positive_remove_host_collection(module_org):
+    """Disassociate host collection from the activation key
+
+    :id: 31992ac4-fe55-45bb-bd17-a191928ec2ab
+
+    :expectedresults:
+
+        1. By default, an activation key is associated with no host
+           collections.
+        2. Associating host collection with activation key add it to the
+           list.
+        3. Disassociating host collection from the activation key actually
+           removes it from the list
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    # An activation key has no host collections by default.
+    act_key = entities.ActivationKey(organization=module_org).create()
+    assert len(act_key.host_collection) == 0
+
+    host_collection = entities.HostCollection(organization=module_org).create()
+
+    # Associate host collection with activation key.
+    act_key.add_host_collection(data={'host_collection_ids': [host_collection.id]})
+    assert len(act_key.read().host_collection) == 1
+
+    # Disassociate host collection from the activation key.
+    act_key.remove_host_collection(data={'host_collection_ids': [host_collection.id]})
+    assert len(act_key.read().host_collection) == 0
+
+
+@pytest.mark.tier1
+def test_positive_update_auto_attach():
+    """Create an activation key, then update the auto_attach
+    field with the inverse boolean value.
+
+    :id: ec225dad-2d27-4b37-989d-1ba2c7f74ac4
+
+    :expectedresults: The value is changed.
+
+    :CaseImportance: Critical
+    """
+    act_key = entities.ActivationKey().create()
+    act_key_2 = entities.ActivationKey(
+        id=act_key.id, auto_attach=(not act_key.auto_attach)
+    ).update(['auto_attach'])
+    assert act_key.auto_attach != act_key_2.auto_attach
+
+
+@pytest.mark.tier1
+@pytest.mark.upgrade
+@pytest.mark.parametrize('name', **parametrized(valid_data_list()))
```

Comment:
Is this needed at all?  I don't see a usage for it and it passes fine without it.

Reply:
forgot to specify the param name in activation key creation, adding in